src/Roslyn.Suppressions/Roslyn.Suppressions.UnitTests/SonarDiagnosticSuppressorTests.cs

```diff
@@ -44,11 +49,137 @@ public void Suppressor_ReturnsExpectedSuppressions()
         public void Suppressors_AllSuppressorsReturnSameInstances()
         {
             // Perf - check every instance reuses the same set of descriptors
-            var supported1 = new SonarDiagnosticSuppressor().SupportedSuppressions;
-            var supported2 = new SonarDiagnosticSuppressor().SupportedSuppressions;
+            var supported1 = CreateTestSubject().SupportedSuppressions;
+            var supported2 = CreateTestSubject().SupportedSuppressions;
 
             // Have to use "Equals" to test identityt equality - see above test.
             supported1.Equals(supported2).Should().BeTrue();
         }
+
+        [TestMethod]
+        public void GetSuppressions_IsNotInConnectedMode_ShouldReturnEmpty()
+        {
+            var suppressionContext = CreateSuppressionContext(null);
+            var reportedDiagnostics = CreateReportedDiagnostics();
+
+            var testSubject = CreateTestSubject();
+
+            var suppressions = testSubject.GetSuppressions(reportedDiagnostics, suppressionContext.Object);
+
+            suppressions.Count().Should().Be(0);
+        }
+
+        [TestMethod]
+        public void GetSuppressions_HasSupressedDiagnostic_ShouldReturnSupressions()
```

Comment:
Minor: repeated typo in this test name and the ones below - "suppressed" has two "p"s.